compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/create_aie_workgroup.mlir

```diff
@@ -0,0 +1,383 @@
+// CHECK:           scf.forall (%{{.*}}, %{{.*}}) in (1, 2)
+func.func @forall() {
+  scf.forall (%arg2, %arg3) in (1, 2) {
+  }
+  return
+}
+
+// -----
+
+// Verify that scf.forall is inserted in both control code and cores.
+//
+// CHECK-LABEL: @forall_cores
+// CHECK:       amdaie.workgroup
+// CHECK:         %[[TILE_0:.+]] = amdaie.tile
+// CHECK:         %[[TILE_1:.+]] = amdaie.tile
+// CHECK:         %{{.+}} = amdaie.core(%[[TILE_0]])
+// CHECK:           scf.forall (%{{.*}}, %{{.*}}) in (1, 2)
+// CHECK:         %{{.+}} = amdaie.core(%[[TILE_1]])
+// CHECK:           scf.forall (%{{.*}}, %{{.*}}) in (1, 2)
+// CHECK:         amdaie.controlcode
+// CHECK:           scf.forall (%{{.*}}, %{{.*}}) in (1, 2)
+func.func @forall_cores() {
+  %c0 = arith.constant 0 : index
+  %c1 = arith.constant 1 : index
+  scf.forall (%arg2, %arg3) in (1, 2) {
+    %tile_0_0 = amdaie.tile(%c0, %c0)
+    %tile_0_1 = amdaie.tile(%c0, %c1)
+    %core_0_0 = amdaie.core(%tile_0_0) {
+      amdaie.end
+    }
+    %core_0_1 = amdaie.core(%tile_0_1) {
+      amdaie.end
+    }
+  }
+  return
+}
+
+// -----
+
+// Verify that empty scf.forall is inserted in control code with nested dmas.
+//
+// CHECK-LABEL: @forall_dmas
+// CHECK:       amdaie.workgroup
+// CHECK-DAG:     %[[FROMMEMREF0:.+]] = amdaie.logicalobjectfifo.from_memref
+// CHECK-SAME:    memref<1x1x8x16xi32, 2> -> !amdaie.logicalobjectfifo<memref<1x1x8x16xi32, 2>>
+// CHECK-DAG:     %[[FROMMEMREF1:.+]] = amdaie.logicalobjectfifo.from_memref
+// CHECK-SAME:    memref<8x16xi32, 1> -> !amdaie.logicalobjectfifo<memref<8x16xi32, 1>>
+// CHECK:         %[[DMA:.+]] = amdaie.circular_dma_cpy_nd
+// CHECK-SAME:    %[[FROMMEMREF0]][] [] []
+// CHECK-SAME:    %[[FROMMEMREF1]][] [] []
+// CHECK-SAME:    (!amdaie.logicalobjectfifo<memref<1x1x8x16xi32, 2>>, !amdaie.logicalobjectfifo<memref<8x16xi32, 1>>)
+// CHECK:         amdaie.controlcode
+// CHECK:           scf.forall (%[[ARG0:.*]], %[[ARG1:.*]]) in (2, 2)
+// CHECK:             %[[IPU_DMA:.+]] = amdaie.npu.dma_cpy_nd %[[DMA]]
+// CHECK-SAME:        [] [] []
+// CHECK-SAME:        [0, 0, 0, 0] [1, 1, 8, 16] [128, %[[ARG1]], %[[ARG0]], 1]
+// CHECK:             amdaie.npu.dma_wait(%[[IPU_DMA]], S2MM)
+func.func @forall_dmas(%arg0: memref<1x1x8x16xi32, 2>, %arg1: memref<8x16xi32, 1>) {
+  %0 = amdaie.logicalobjectfifo.from_memref %arg0, {} : memref<1x1x8x16xi32, 2> -> !amdaie.logicalobjectfifo<memref<1x1x8x16xi32, 2>>
+  %1 = amdaie.logicalobjectfifo.from_memref %arg1, {} : memref<8x16xi32, 1> -> !amdaie.logicalobjectfifo<memref<8x16xi32, 1>>
+  scf.forall (%arg2, %arg3) in (2, 2) {
+    %2 = amdaie.dma_cpy_nd(%0[] [] [], %1[0, 0, 0, 0] [1, 1, 8, 16] [128, %arg3, %arg2, 1]) : (!amdaie.logicalobjectfifo<memref<1x1x8x16xi32, 2>>, !amdaie.logicalobjectfifo<memref<8x16xi32, 1>>)
+  }
+  return
+}
+
+// -----
+
+// CHECK-LABEL: @func
+// CHECK:       amdaie.workgroup
+// CHECK:         amdaie.controlcode
+func.func @func() {
+  return
+}
```

Comment:
Nit: This lit test specifically should be moved to the top.

Reply:
Yeah, I wanted to order the tests lexicographically, but didn't fully adhere to this anyway as I put the `complex_example` at the bottom. So, I moved this to the top as suggested.